cmd/dockerd/malformed_host_override.go

```diff
@@ -0,0 +1,80 @@
+// +build !windows
+
+package main
```

Comment:
I'm actually writing some basic test (leaving this is the main package) but yeah moving it to a separate pkg may be good. Did you write benchmarks for this PR?